docs/src/test/java/sample/javadsl/RunnableGraphStreamWrapperActor.java

```diff
@@ -0,0 +1,194 @@
+                .withBootstrapServers("localhost:9092")
+                .withGroupId("group1")
+                .withProperty(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+
+        producerSettings = ProducerSettings.create(system, new StringSerializer(), new StringSerializer())
+                .withBootstrapServers("localhost:9092")
+                .withProperty("compression.type", "lz4")
+                .withProperty("acks", "all");
+    }
+
+    public void onReceive(Object messageObj) {
+        // Trigger the pipeline on receive a message
+        pipeline();
+    }
+
+    /**
+     * Reactive Kafka stream.
+     */
+    private void pipeline() {
+        ActorMaterializer materializer = ActorMaterializer.create(ActorMaterializerSettings
+                .create(system)
+                .withDebugLogging(true)
+                .withSupervisionStrategy(stopOnSpecificExceptionDecider(getSelf())), system);
+
+        // Source: Receive messages and transform to a EventMessage
+        Source<EventMessage, Consumer.Control> sourceStream = Consumer.committableSource(consumerSettings, Subscriptions.topics("topic1"))
+                .map(committableMessage -> new EventMessage(committableMessage, committableMessage.record().value()));
+
+        // Flow to process the message
+        Flow<EventMessage, EventMessage, NotUsed> firstPartitionFlow = Flow.of(EventMessage.class).map(this::processMessage); // Save and Transform
+        Flow<EventMessage, EventMessage, NotUsed> secondPartitionFlow = Flow.of(EventMessage.class).map(this::processMessage); // Delete and Transform
+
+        // Flow to create Kafka ProducerMessage
+        Flow<EventMessage, ProducerMessage.Message<String, String, ConsumerMessage.Committable>, NotUsed> produceMessageFlow = Flow.of(EventMessage.class)
+                .map(eventTuple -> produceCommittableEvent(eventTuple, "producerTopic"));
+
+        // Sink: Committable producer sink
+        Sink<ProducerMessage.Message<String, String, ConsumerMessage.Committable>, CompletionStage<Done>> producerCommitSink = Producer.commitableSink(producerSettings);
+        // Simple commit the offset!
+        Sink<EventMessage, CompletionStage<Done>> commitOnlySink = Sink.foreach(eventTuple -> eventTuple.committableMessage.committableOffset().commitJavadsl());
+
+        Graph<UniformFanOutShape<EventMessage, EventMessage>, NotUsed> eventTypePartition = Partition.create(3, (Function<EventMessage, Object>) tuple -> {
+            switch (tuple.message) {
+                case "EVENT_TYPE_1":
+                    return 1;
+                case "EVENT_TYPE_2":
+                    return 2;
+                default:
+                    return 0;
+            }
+        });
+
+        // Build the Partition Graph
+        Graph<ClosedShape, Consumer.Control> completionStageGraph = GraphDSL.create(sourceStream, (builder, sourceShape) -> {
```

Comment:
I thought about it, however I felt, it would be good to cover comprehensive example so it might help others! Reduced the code little.